### ISSUE TITLE
select & cursor jumps back to the beginning of the text

### ISSUE THREAD

**Issue by HI54U** [billthefarmer/editor#56]
First, thank you for your lite and very handy editor, which I am using with Los15.1 on my smartphone!

If I open a text file where I want just to copy one line i.e. on the 8th page of the text file, the cursor will go straight back to the beginning of the text file and I need to scroll down again to the page where I want to copy a word/number.

This is due the text file is opened as read only which is good (related to #33), because I don't want to change the text, I just want to select a part and want to copy the text.

Is there any possibility to select text on a page far down, that the cursor remains on the selected part and you don't need to scroll back from the beginning of the text?

**Comment by billthefarmer** [comment]
I don't understand the 'again'. If you open a text file, scroll down, open another app, and then go back to the editor it will still be in the same position. If you open the text file again, it may scroll to the saved position, depending on whether you did something in the editor to cause it to save the position. Any menu item causes the position to be saved, including the back button. However the back button closes the app, so if you go back to it, it opens the default file.

**Comment by HI54U** [comment]
thank you for the quick response. I'll try to explain the steps. I open a text file from the editor. I scroll to the page where I want to select a word and copy it. Then the cursor jumps always back to to beginning of the text file. So I need to scroll again down to the word. Tested this with different text files and also with other editors where I can open a text file and then select a word where ever it is in the file and just select it and can copy it without that the cursor jumps to the beginning of the text file. Hope it's clearer explained what I  want to say.

**Comment by billthefarmer** [comment]
Ah, got it. I think it must be due to the workaround I put in in response to #33, which temporarily changes the text size, which will possibly cause the position to change. If so, I will make the workaround android 6 specific.

**Comment by billthefarmer** [comment]
OK, I've released a new version which changes the typeface rather than the size and only in android 6. That doesn't move the cursor.

**Comment by HI54U** [comment]
Dear Bill, 1st: thank you for your quick fix. The cursor stays now where I want to select a text.
But now I am not able to select any text/numbers/words. This isn't working in version 1.24.

**Comment by billthefarmer** [comment]
See #56. I am working on a new version.

**Comment by billthefarmer** [comment]
When I originally wrote this app it was intended for editing small files, like a bit of CSS or JavaScript. I never really tested it on files more than a page long. The default behavior of the text view, I have found, when you switch to edit mode is to scroll so that the last cursor position (current selection) is visible. That will be the end of the file, or the beginning, or the saved position if there is one. So to get around that I have set the position to the middle of the current display, so that it doesn't scroll when you switch. That may not be where it was left if the file had been previously edited.
The new version has been published by F-Droid.
